WordPress/wporg-main-2022
# Add RTLCSS support

Fixes #124. We need to be able to run `rtlcss` on the blocks (so, during `wp-scripts build` & `wp-scripts start`), and on the CSS in `style.css`. Ideally, these would be the same process. We can add an `rtlcss` step to wp-scripts with a custom webpack config.

Instead of using the regular theme's `style.css`, I've moved the CSS into the `src` directory as another "fake block" (like the download modal). This way, we can use the same build step for everything. This is admittedly a kind-of hacky workaround, and it requires an empty JS file that isn't used anywhere.

But it works — with the added webpack config, all the CSS files also generate RTL versions.

### Screenshots

There aren't many CSS differences to point to, but you can see the difference in the features list and in the download modal.

| Before | After |
|--------|-------|
| ![before-css](https://user-images.githubusercontent.com/541093/190269395-566542d4-1804-4ad5-8aaf-9720cf60b625.png) | ![after-css](https://user-images.githubusercontent.com/541093/190269393-c12af804-f241-4ac7-9594-e7f0c7de72cb.png) |
| <img width="624" alt="before-modal" src="https://user-images.githubusercontent.com/541093/190269907-82033055-52b7-4fec-b8ef-81864a8d9090.png"> | <img width="636" alt="after-modal" src="https://user-images.githubusercontent.com/541093/190269906-ce6d4ba8-0929-493c-b580-2fb2eb9e4e5c.png"> |

### How to test the changes in this Pull Request:

1. Run `yarn workspace wporg-main-2022-theme build`
2. You should see new files in `build`, a new folder called `styles` & new CSS files in the `download` folder.
